app/code/core/Mage/ConfigurableSwatches/Helper/Productimg.php

```diff
@@ -343,7 +343,7 @@ public function filterImageInGallery($product, $image)
         }
 
         if (!isset($this->_productImageFilters[$product->getId()])) {
-            $mapping = call_user_func_array("array_merge_recursive", $product->getChildAttributeLabelMapping());
+            $mapping = call_user_func_array("array_merge_recursive", array_values($product->getChildAttributeLabelMapping()));
             $filters = array_unique($mapping['labels']);
```

Comment:
Notice: Undefined index: labels in /app/code/core/Mage/ConfigurableSwatches/Helper/Productimg.php on line 347